Initial commit for em4x70 support.

cmolson opened the pull request:
Initially, I only have an em4170 variant used for car transponders.
Also known as the ID48 and possibly Magic II.

This first feature is to read the tag id and user memory 1 and 2.

iceman1001 commented on client/src/cmdlfem.c:
```diff
@@ -25,6 +27,7 @@ static command_t CommandTable[] = {
     {"410x",  CmdLFEM410X,  AlwaysAvailable, "EM 4102 commands..."},
     {"4x05",  CmdLFEM4X05,  AlwaysAvailable, "EM 4205 / 4305 / 4369 / 4469 commands..."},
     {"4x50",  CmdLFEM4X50,  AlwaysAvailable, "EM 4350 / 4450 commands..."},
+    {"4x70",  CmdLFEM4X70,  AlwaysAvailable, "EM 4170 /4070 commands..."},
```
swap order,  and space

iceman1001 commented on client/src/cmdlfem410x.c:
```diff
@@ -10,6 +10,7 @@
 
 #include "cmdlfem410x.h"
 #include "cmdlfem4x50.h"
+#include "cmdlfem4x70.h"
```
I don't think it belongs here

cmolson replied:
Right!

iceman1001 commented on client/src/cmdlfem4x70.c:
```diff
@@ -0,0 +1,151 @@
+
+    em4x70_data_t etd = {0};
+
+    while (param_getchar(Cmd, cmdp) != 0x00 && !errors) {
```
use the cliparser

iceman1001 commented on include/em4x70.h:
```diff
@@ -0,0 +1,18 @@
+
+typedef struct {
+    bool parity;
+} PACKED em4x70_data_t;
```
packed struct with a bool??

cmolson replied:
I sort of copied the em4x50_data_t struct, I removed the items I wasn't using yet. I have added more items for the next pull request that includes writing support.

I'll remove the PACKED attribute in this push.

iceman1001 replied:
It just didn't make any sense to have an struct with a bool inside.  So its for next PRs  ok.

iceman1001 commented on armsrc/em4x70.c:
```diff
@@ -0,0 +1,594 @@
+        bits+=8;
+        Dbprintf("Read: %02X", bytes[i]);
+    }
+}*/
```
remove custom debug code.
there exists already a few  (see appmain.c)

iceman1001 reviewed (changes requested):
only convert to cliparser and remove your custom debug stuff and it looks fine

iceman1001 commented:
And you will need to resolve the conflicting file

cmolson commented:
I will try and add the cliparser change today as well. Thanks!

I named this em4x70 as the transponder/tag I am using is very close to that, with the only difference from what I can see is in the parity of the commands. I will be continuing to update this and have a few more PR's to submit once this large one is in place.

The transponder/tag I am using is referred to ID48 in the car world.

iceman1001 commented:
yeah,  we might need to rename it later on but for now it will do.
You should mention ID48 inside the help text of your `info` command in order to make things clear.

cmolson commented:
I've updated to use the cliparser. I also added some text about ID48 automotive transponders to the help text.

There was also a small bugfix to the regular EM4x70 devices that require parity bit in the command. We need to drop the most significant bit of the command before adding a parity so that the entire length remains 4 bits.

iceman1001 commented:
As a initial PR,  its looks ok,   most likely some minor things but we sort it out.   There was a spelling mistake in the help text among others.

Thanks for the PR and the new supported type!
